### PR TITLE
Make the auto-discovery kubelet listener look for docker image in the spec

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

Make the auto-discovery kubelet listener look for docker image name in the pod spec rather than in the pod status.

### Motivation

This fixes unexpected AD behaviours when when several pods with different docker image name that are pointing to the same image digest are running on the same node.

If, on a node, several docker image names are aliasing the same docker image digest, the image name that is reported by Kubernetes in the pod status section can be any of those names.
On the other hand, the image name that appears in the pod spec section is the name that was entered by the user.

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

* Spawn an EKS cluster
* Create two docker images with two different names that are pointing to the same digest (`docker tag <IMAGE1> <IMAGE2>`)
* Spawn two pods on the same node, using the two docker image name.
* Create an AD config that matches only one of the two docker image names.

With the old version of the agent, we would see either two or zero checks scheduled.
With this change, we should see only one check scheduled.

### CODE REVIEW

**Review by vboulineau — APPROVED** [review]

**Comment by vboulineau** [comment]
Just double checking: no way to unit test this?

File: releasenotes/notes/Look-for-image-in-spec-rather-than-status-95d2402b82068d1c.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Make the auto-discovery kubelet listener look for docker image name
+    in the pod spec rather than in the pod status.
+    This fixes unexpected AD behaviors when when several pods with different
+    docker image name that are pointing to the same image digest are
+    running on the same node.
```
Comment (ahmed-mez) [inline releasenotes/notes/Look-for-image-in-spec-rather-than-status-95d2402b82068d1c.yaml:15]: can we reword this to be easier for customers to understand? we're fixing a bug that (in certain conditions) was causing not picking checks and logs for containers targeted by container-image-based autodiscovery
